src/pages/auth/hooks/useEmailPrompt.ts

```diff
@@ -0,0 +1,77 @@
+import { yupResolver } from '@hookform/resolvers/yup'
+import { atom, useAtomValue, useSetAtom } from 'jotai'
+import { atomWithStorage } from 'jotai/utils'
+import { useEffect, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import * as yup from 'yup'
+
+import { useAuthContext } from '../../../context'
+import { useUserEmailUpdateMutation } from '../../../types'
+import { authUserAtom } from '../state'
+
+type FormValues = Record<string, any>
+
+const emailSchema = yup.object().shape({
+  dontAskAgain: yup.boolean(),
+  email: yup
+    .string()
+    .email('Invalid email')
+    .when('dontAskAgain', {
+      is: false,
+      then: (schema) => schema.required('Email is required'),
+      otherwise: (schema) => schema.notRequired(),
+    }),
+})
+
+export const dontAskAgainAtom = atomWithStorage('dontAskAgain', false)
+
+export const shouldPromptAtom = atom((get) => {
+  const user = get(authUserAtom)
+  const dontAskAgain = get(dontAskAgainAtom)
+  return user.id && !user.email && !dontAskAgain
+})
```

Comment:
Try to export non JSX stuff from a different file, this is not necessary, but I did some research recently, and having JSX and non JSX stuff exported with named exports from the same file is one of the blocker for React Hot-reload. 

As you must have realized, we have to manually refresh almost every time. 

There are a few constraints for that, apart from the one I mentioned above, the other few ones are:
1. no` export * from  '../thisthat'` instead export each item, like `export {this, that} from '../this that'`

I've been trying to implement this in the redesign. Still haven't gotten there though. Just giving a headsup.